arch/arm/src/arm/arm_backtrace.c

```diff
@@ -0,0 +1,268 @@
+/****************************************************************************
+ * arch/arm/src/arm/arm_backtrace.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <nuttx/arch.h>
+
+#include "sched/sched.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Macro and definitions for simple decoding of instuctions.
+ * To check an instruction, it is ANDed with the IMASK_ and
+ * the result is compared with the IOP_. The macro INSTR_IS
+ * does this and returns !0 to indicate a match.
+ */
+
+#define INSTR_IS(i, o) (((i) & (IMASK_##o)) == (IOP_##o))
+
+#define IMASK_T_STMDB     0xfffff000  /* stmdb sp!,{..lr} */
+#define IOP_T_STMDB       0xe92d4000
+
+#define IMASK_T_PUSH_LO   0xff00      /* push {reglist} (not LR) */
+#define IOP_T_PUSH_LO     0xb400
+
+#define IMASK_T_PUSH      0xff00      /* push {reglist} (inc LR) */
+#define IOP_T_PUSH        0xb500
+
+#define IMASK_T_VPUSH_16  0xffbf8f00  /* vpush d */
+#define IOP_T_VPUSH_16    0xed2d8b00
+
+#define IMASK_T_VPUSH_8   0xffbf8f00  /* vpush s */
+#define IOP_T_VPUSH_8     0xed2d8a00
+
+#define IMASK_T_SUB_SP_16 0xff80      /* sub sp, # */
+#define IOP_T_SUB_SP_16   0xb080
+
+#define IMASK_T_SUB_SP_32 0xfbff8f00  /* sub.w sp, sp, # */
+#define IOP_T_SUB_SP_32   0xf1ad0d00
+
+#define INSTR_LIMIT       0x2000
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: getlroffset
```

Comment:
Please include a complete Description of the function, including params and return